plugins/chat/app/services/chat/action/publish_auto_removed_user.rb

```diff
@@ -8,29 +8,26 @@ module Action
     # this in staff actions so it's obvious why these users were
     # removed.
     class PublishAutoRemovedUser < Service::ActionBase
-      # @param [Symbol] event_type What caused the users to be removed,
+      # @param [Symbol] event What caused the users to be removed,
       #   each handler will define this, e.g. category_updated, user_removed_from_group
       # @param [Hash] users_removed_map A hash with channel_id as its keys and an
       #   array of user_ids who were removed from the channel.
-      option :event_type
+      option :event
       option :users_removed_map
 
       def call
         return if users_removed_map.empty?
 
-        users_removed_map.each do |channel_id, user_ids|
+        users_removed_map.each do |channel_id, all_user_ids|
           job_spacer = JobTimeSpacer.new
-          user_ids.in_groups_of(1000, false) do |user_id_batch|
-            job_spacer.enqueue(
-              Jobs::Chat::KickUsersFromChannel,
-              { channel_id: channel_id, user_ids: user_id_batch },
-            )
+          all_user_ids.in_groups_of(1000, false) do |user_ids|
+            job_spacer.enqueue(Jobs::Chat::KickUsersFromChannel, { channel_id:, user_ids: })
```

Comment:
@martin-brennan Do you remember why we use `JobTimeSpacer` here which creates multiple messageBus messages instead of just sending a single messageBus message? Note that the `user_ids` even if large are never sent to the client, it is only used to determine if the user should receive the message on the server side.

Reply:
From what I remember this is to stop the client from churning tons of updates if tons of users get kicked from their channel. There is some discussion about it in /t/70607/35 internally. E.g. if 20000 users were kicked from a channel at once, they would need to navigate to another channel and load messages there or some other part of the site and cause more AJAX requests. I remember discussing this with @jjaffeux at the time

Reply:
Also this is in the original PR description https://github.com/discourse/discourse/pull/20344  :

![image](https://github.com/user-attachments/assets/43c73bd5-dda7-4fbe-92d6-c47fdd13b381)

Reply:
Also this is in the original PR description https://github.com/discourse/discourse/pull/20344  :

![image](https://github.com/user-attachments/assets/43c73bd5-dda7-4fbe-92d6-c47fdd13b381)

Reply:
> E.g. if 20000 users were kicked from a channel at once, they would need to navigate to another channel and load messages there or some other part of the site and cause more AJAX requests. I remember discussing this with @jjaffeux at the time

I see and this assumes that 20K users are active at the same time. If 20K users are active at the same thing, the load on the server is already there so I feel like we don't really need to space stuff out. If don't have to space stuff out, we could reduce the load on Sidekiq since we can just publish a single message. @ZogStriP Would this help reduce the thundering herd of Sidekiq jobs we are seeing internally?

Reply:
I debated changing that one as well but didn't want to make this PR larger than it needs to be. I'm more than happy to do the work in another PR though.